runtime/Java/src/org/antlr/v4/runtime/atn/ATNSerializer.java

```diff
@@ -286,86 +280,129 @@ public void serializeCodePoint(IntegerList data, int cp) {
 				data.add(arg3);
 			}
 		}
+	}
 
-		int ndecisions = atn.decisionToState.size();
-		data.add(ndecisions);
-		for (DecisionState decStartState : atn.decisionToState) {
-			data.add(decStartState.stateNumber);
+	private Map<IntervalSet, Integer> addSets() {
+		List<IntervalSet> bmpSets = new ArrayList<>();
+		List<IntervalSet> smpSets = new ArrayList<>();
```

Comment:
These code becomes useless since all integers becomes 32-bit. It should be simplified (no distinguish between bmd and smp sets).

Reply:
yep! That's next on my list.

Reply:
ok, I think i've pushed all 32-bit stuff minus Java target.  All sets are 32-bit ranges not 16 and 32 (bmp,smp).